nucleus/common/internal-api/src/main/java/org/glassfish/internal/data/ApplicationInfo.java

```diff
@@ -112,12 +112,12 @@ private void createServiceLocator() {
 
         appServiceLocator = slf.create(locatorName);
         deploymentFailedListener = new DeploymentFailedListener(source);
-        events.register(deploymentFailedListener);
+        unregisterEventListener(deploymentFailedListener);
```

Comment:
Maybe we should **register** listener here instead of *unregister*?

Reply:
Do you have any idea why it did not break any test? Whole TCK passed, I am just locally running just few remaining which failed because of read timeouts caused by my slow network.
That is really stupid mistake I made. I do wonder how I did that ... thank you for noticing that!

Reply:
`DeploymentFailedListener` calls `disposeServiceLocator()` method. This method unregister the listener itself and destroy the application service locator in case of a deployment failure. I think, we just have a memory leak in this case.